entity-service-impl/src/test/java/org/hypertrace/entity/query/service/converter/UpdateConverterTest.java

```diff
@@ -118,8 +119,18 @@ void testAllCombinations(
         .thenReturn(Optional.of("attributes.subDocPath"));
     when(mockEntityAttributeMapping.getAttributeKind(requestContext, "columnName"))
         .thenReturn(Optional.ofNullable(attributeKind));
-    when(mockEntityAttributeMapping.isMultiValued(attributeKind))
-        .thenReturn(valueType == STRING_ARRAY);
+    when(mockEntityAttributeMapping.isArray(attributeKind)).thenReturn(valueType == STRING_ARRAY);
+    if (operator.equals(ATTRIBUTE_UPDATE_OPERATOR_SET) && !valueType.equals(STRING_ARRAY)) {
+      when(mockEntityAttributeMapping.isPrimitive(attributeKind))
```

Comment:
Using `when.. then... return` inside an `if` condition should be avoided (as our testing code should contain minimal conditional logic).

Reply:
Was getting unnecessary stubbing for some of the configurations thats why used `if` condition. Split the test into 2 tests with appropriate stubbings.